### PR TITLE
Fix/side attacker positioning

### PR DESCRIPTION
**Pull request by RobZelluf**
- Improve the claiming of positions by the side attackers.
- Also improve passing and receiving a bit
- Let the attacker get behind the ball better, instead of going through it

### CODE REVIEW

File: roboteam_ai/src/coach/OffensiveCoach.cpp [hunk 1]
```diff
@@ -111,6 +111,29 @@ std::vector<Vector2> OffensiveCoach::getOffensivePositionVectors() {
     return positionVectors;
 }
 
+const vector<OffensiveCoach::RobotPtr> &OffensiveCoach::getSideAttackers() const {
+    return sideAttackers;
+}
+
+void OffensiveCoach::addSideAttacker(OffensiveCoach::RobotPtr robot) {
+    for(auto &sideAttacker : sideAttackers) {
```
Comment (Barisimre) [inline roboteam_ai/src/coach/OffensiveCoach.cpp]: why not just use a set? But I guess if it works it works

File: roboteam_ai/src/skills/Attack.cpp [hunk 2]
```diff
@@ -17,57 +17,54 @@ Attack::Attack(string name, bt::Blackboard::Ptr blackboard)
...
     shot = false;
 }
 
 /// Get an update on the skill
 bt::Node::Status Attack::onUpdate() {
     if (! robot) return Status::Running;
 
-    if (shot && !world::world->robotHasBall(robot->id, world::OUR_ROBOTS)) {
+    if (shot && !world::world->ourRobotHasBall(robot->id)) {
+        std::cout << "SUCCESS" << std::endl;
```
Comment (Barisimre) [inline roboteam_ai/src/skills/Attack.cpp]: Remove print

File: roboteam_ai/src/skills/Attack.cpp [hunk 3]
```diff
@@ -17,57 +17,54 @@ Attack::Attack(string name, bt::Blackboard::Ptr blackboard)
...
+        targetPos = ball->pos;
+        pva = basicGtp.getPosVelAngle(robot, targetPos);
+        command.w = (world::field->get_their_goal_center() - ball->pos).toAngle().getAngle();
+        std::cout << world::field->get_their_goal_center() << " - " << ball->pos << " - " << command.w << std::endl;
         if (world::world->robotHasBall(robot->id, true, Constants::MAX_KICK_RANGE())) {
             command.kicker = 1;
             command.kicker_vel = static_cast<float>(rtt::ai::Constants::MAX_KICK_POWER());
             command.kicker_forced = 1;
             shot = true;
+            std::cout << "SHOT" << std::endl;
```
Comment (Barisimre) [inline roboteam_ai/src/skills/Attack.cpp]: Remove print

File: roboteam_ai/src/skills/Receive.cpp [hunk 4]
```diff
@@ -50,12 +55,13 @@ Receive::Status Receive::onUpdate() {
     if (coach::g_pass.isPassed()) {
         // Remember the status of the ball at the moment of passing
         if(!isBallOnPassedSet) {
+            std::cout << robot->id << " - " << ball->vel << std::endl;
```
Comment (Barisimre) [inline roboteam_ai/src/skills/Receive.cpp]: print

File: roboteam_ai/src/skills/Receive.cpp [hunk 5]
```diff
@@ -141,14 +147,20 @@ void Receive::intercept() {
 bool Receive::passFailed() {
     //TODO: Remove print statements and make 1 big if statement
     if ((ball->vel.toAngle() - ballOnPassed->vel.toAngle()).getAngle() > 0.5) {
+        std::cout << "Ball deflected!" << std::endl;
```
Comment (Barisimre) [inline roboteam_ai/src/skills/Receive.cpp]: print

File: roboteam_ai/src/skills/Receive.cpp [hunk 6]
```diff
@@ -141,14 +147,20 @@ void Receive::intercept() {
...
         return true;
     }
 
-    return receiverMissedBall();
+    if (receiverMissedBall()) {
+        std::cout << "Receiver missed ball" << std::endl;
+        return true;
+    }
+
+    return false;
```
Comment (Barisimre) [inline roboteam_ai/src/skills/Receive.cpp]: Okay just use
```
return receiverMissedBall();
```

**Review by Barisimre — CHANGES_REQUESTED** [review]

**Review by rolfvdhulst — CHANGES_REQUESTED** [review]

**Review by rolfvdhulst — APPROVED** [review]

**Review by Barisimre — APPROVED** [review]

**Pull request merged**